app/components/edit/tab_form/tab_list_component.rb

```diff
@@ -4,16 +4,19 @@ module Edit
   module TabForm
     # Component for rendering tabbed navigation for the provided form.
     class TabListComponent < ApplicationComponent
-      renders_many :tabs, 'Edit::TabForm::TabComponent'
+      renders_many :tabs, Edit::TabForm::TabComponent
       # The reason for the before_form_panes is to
       # avoid nested form tags, which aren't allowed in HTML.
       # Note that to reference the main form from elements in the before_form_pane,
       # provide a form_id to this component.
       # The form_id can then be provided as the form attribute to an input element to
       # connect it to the main form.
       # This approach is used to allow submit buttons to be outside the main form.
-      renders_many :before_form_panes
-      renders_many :panes
+      renders_many :before_form_panes, Works::Edit::PaneComponent
+      renders_many :panes, types: {
+        work: { renders: Works::Edit::PaneComponent, as: :work_pane },
+        collection: { renders: Collections::Edit::PaneComponent, as: :collection_pane }
+      }
```

Comment:
Being more explicit about which components these slots expect has trade-offs. The downside is that we are temporarily coupled to using these components in these slots. The upside is less machinery when creating panes. See below for examples of the payoff.

Reply:
This is totally reasonable since this component is specifically for work and collections (i.e., it is not generic).